backend/app/main.py

```
@@ -85,10 +85,15 @@ async def lifespan(app: FastAPI):
 app.add_middleware(RequestLoggingMiddleware)
 app.add_middleware(AuthenticationMiddleware)
 
-# Add rate limiting middleware with testing bypass
+# Add rate limiting middleware with conditional testing bypass
 import os
-if not (os.getenv('TESTING') == 'true' or os.getenv('CI') == 'true'):
+# Only disable rate limiting if explicitly requested, not just because we're in CI/CD
+## WARNING: Disabling rate limiting exposes the API to abuse and denial of service attacks.
+# Only disable rate limiting in controlled testing environments. NEVER disable in production!
+if not (os.getenv('DISABLE_RATE_LIMITING') == 'true'):
```

Ruff (0.11.9)

Use `os.getenv('DISABLE_RATE_LIMITING') != 'true'` instead of `not os.getenv('DISABLE_RATE_LIMITING') == 'true'`

Replace with `!=` operator

(SIM201)